crates/router/src/core/user.rs

```diff
@@ -71,6 +71,38 @@ pub async fn signup_with_merchant_id(
     }))
 }
 
+pub async fn get_user_details(
+    state: AppState,
+    user_from_token: auth::UserFromToken,
+) -> UserResponse<user_api::GetUserDetailsResponse> {
+    let user = state
+        .store
+        .find_user_by_id(&user_from_token.user_id)
+        .await
+        .map_err(|e| {
+            if e.current_context().is_db_not_found() {
+                e.change_context(UserErrors::InvalidCredentials)
+            } else {
+                e.change_context(UserErrors::InternalServerError)
+            }
+        })?
```

Comment:
This should be `InternalServerError` always as we are getting the `user_id` from the token.

Reply:
See next comment before changing this.